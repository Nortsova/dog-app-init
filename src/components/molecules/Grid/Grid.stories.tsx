import type { Meta, StoryObj } from '@storybook/react';
import { Grid } from '.';

const meta = {
  title: 'Molecules/Grid',
  component: Grid
} satisfies Meta<typeof Grid>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {}
};
