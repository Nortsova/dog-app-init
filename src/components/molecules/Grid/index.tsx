import { Box, IconButton, styled } from '@mui/material';
import { FC, useState } from 'react';
import { Order } from '../../../services/images';
import { CardItem } from '../../atoms/CardItem';
import { IconHeart } from '../../atoms/IconHeart';
import { useAddFavouritesMutation } from '../../../services/favourites';
import { useGetImagesWithFavourites } from '../../../hooks';

type GridProps = {
  page: number;
  order: Order;
};

const StyledBox = styled(Box)({
  display: 'grid',
  gridTemplateColumns: 'repeat(4, 275px)',
  gridAutoRows: '196px',
  gridGap: '20px',
  width: '1160px',
  marginLeft: 'auto',
  marginRight: 'auto',
  '& .MuiBox-root:first-of-type': {
    gridColumn: '1 / 3',
    gridRow: '1 / 3'
  },
  '& .MuiBox-root:nth-of-type(2)': {
    gridColumn: '3',
    gridRow: '1'
  },
  '& .MuiBox-root:nth-of-type(3)': {
    gridColumn: '4',
    gridRow: '1 / 3'
  },
  '& .MuiBox-root:nth-of-type(4)': {
    gridColumn: '1',
    gridRow: '3 / 5'
  },
  '& .MuiBox-root:nth-of-type(5)': {
    gridColumn: '2',
    gridRow: '3'
  },
  '& .MuiBox-root:nth-of-type(6)': {
    gridColumn: '3',
    gridRow: '2 / 4'
  },
  '& .MuiBox-root:nth-of-type(7)': {
    gridColumn: '4',
    gridRow: '3'
  },
  '& .MuiBox-root:nth-of-type(8)': {
    gridColumn: '2',
    gridRow: '4'
  },
  '& .MuiBox-root:nth-of-type(9)': {
    gridColumn: '3',
    gridRow: '4'
  },
  '& .MuiBox-root:nth-of-type(10)': {
    gridColumn: '4',
    gridRow: '4'
  }
});

export const Grid: FC<GridProps> = ({ page, order }) => {
  const [addFavourite] = useAddFavouritesMutation();
  const { data: favouritesImages, isLoading } = useGetImagesWithFavourites({
    page,
    order
  });
  const [isHoveredCard, setIsHoveredCard] = useState<number | null>(null);

  const handleBtnAdd = (id: string) => () => {
    addFavourite({ image_id: id });
  };

  return (
    <StyledBox>
      {isLoading && <div>Loading...</div>}
      {(favouritesImages || []).map((item, index) => {
        const isHover = isHoveredCard === index;

        return (
          <CardItem
            key={item.id}
            variant="primary"
            borderRadius={0}
            onMouseOver={() => setIsHoveredCard(index)}
            onMouseOut={() => setIsHoveredCard(null)}
          >
            <img src={item.url} alt={item.id} height="100%" />
            {item.isFavourite ? (
              <IconButton
                sx={{
                  position: 'absolute',
                  top: '10px',
                  right: '10px'
                }}
                onClick={handleBtnAdd(item.id)}
              >
                <IconHeart state="active" />
              </IconButton>
            ) : (
              <IconButton
                sx={{
                  position: 'absolute',
                  top: '10px',
                  right: '10px'
                }}
                onClick={handleBtnAdd(item.id)}
              >
                {isHover ? (
                  <IconHeart state="hover" />
                ) : (
                  <IconHeart state="default" />
                )}
              </IconButton>
            )}
          </CardItem>
        );
      })}
    </StyledBox>
  );
};
