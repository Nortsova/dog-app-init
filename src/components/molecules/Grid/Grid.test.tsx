import { composeStories } from '@storybook/react';
import * as stories from './Grid.stories';
import { testSnapshot } from '../../../test/utils/snapshot.utils';

describe('NavBar', () => {
  const { Primary } = composeStories(stories);
  testSnapshot(<Primary />);
});
