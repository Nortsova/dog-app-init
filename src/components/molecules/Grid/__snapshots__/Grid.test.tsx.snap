// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid should match snapshot Component 1`] = `
.emotion-0 {
  display: grid;
  grid-template-columns: repeat(4, 275px);
  grid-auto-rows: 196px;
  grid-gap: 20px;
  width: 1160px;
  margin-left: auto;
  margin-right: auto;
}

.emotion-0 .MuiBox-root:first-of-type {
  grid-column: 1/3;
  grid-row: 1/3;
}

.emotion-0 .MuiBox-root:nth-of-type(2) {
  grid-column: 3;
  grid-row: 1;
}

.emotion-0 .MuiBox-root:nth-of-type(3) {
  grid-column: 4;
  grid-row: 1/3;
}

.emotion-0 .MuiBox-root:nth-of-type(4) {
  grid-column: 1;
  grid-row: 3/5;
}

.emotion-0 .MuiBox-root:nth-of-type(5) {
  grid-column: 2;
  grid-row: 3;
}

.emotion-0 .MuiBox-root:nth-of-type(6) {
  grid-column: 3;
  grid-row: 2/4;
}

.emotion-0 .MuiBox-root:nth-of-type(7) {
  grid-column: 4;
  grid-row: 3;
}

.emotion-0 .MuiBox-root:nth-of-type(8) {
  grid-column: 2;
  grid-row: 4;
}

.emotion-0 .MuiBox-root:nth-of-type(9) {
  grid-column: 3;
  grid-row: 4;
}

.emotion-0 .MuiBox-root:nth-of-type(10) {
  grid-column: 4;
  grid-row: 4;
}

<div>
  <div
    class="MuiBox-root emotion-0"
  >
    <div>
      Loading...
    </div>
  </div>
</div>
`;
